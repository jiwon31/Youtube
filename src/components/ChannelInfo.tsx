import { useQuery } from "@tanstack/react-query";
import { useYoutubeApi } from "context/YoutubeApiContext";

type ChannelInfoProps = {
  id: string;
  name: string;
};

export default function ChannelInfo({ id, name }: ChannelInfoProps) {
  const { youtube } = useYoutubeApi();
  const { data: url } = useQuery(
    ["channel", id],
    async () => youtube.channelImageURL(id),
    { staleTime: 1000 * 60 * 5 }
  );

  return (
    <div className="flex items-center my-4 mb-8">
      {url && (
        <img className="w-10 h-10 rounded-full mr-2" src={url} alt={name} />
      )}
      <p className="text-lg font-medium">{name}</p>
    </div>
  );
}
