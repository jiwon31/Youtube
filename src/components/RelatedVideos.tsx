import { useQuery } from "@tanstack/react-query";
import { useYoutubeApi } from "context/YoutubeApiContext";
import VideoCard from "./VideoCard";

type RelatedVideosProps = { id: string };

export default function RelatedVideos({ id }: RelatedVideosProps) {
  const { youtube } = useYoutubeApi();
  const {
    isLoading,
    error,
    data: videos,
  } = useQuery(["related", id], async () => youtube.relatedVideos(id), {
    staleTime: 1000 * 60 * 5,
  });

  return (
    <>
      {isLoading && <p>Loading...</p>}
      {error && <p>Something is wrong ðŸ˜£</p>}
      <ul>
        {videos &&
          videos.map((video) => (
            <VideoCard key={video.id} video={video} type="list" />
          ))}
      </ul>
    </>
  );
}
