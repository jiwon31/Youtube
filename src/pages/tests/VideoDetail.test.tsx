import ChannelInfo from "components/ChannelInfo";
import RelatedVideos from "components/RelatedVideos";
import VideoDetail from "pages/VideoDetail";
import renderer from "react-test-renderer";
import { Route } from "react-router-dom";
import { withRouter } from "tests/utils";
import { fakeVideo } from "tests/videos";

jest.mock("../../components/ChannelInfo");
jest.mock("../../components/RelatedVideos");

const ChannelInfoMock = ChannelInfo as jest.Mock;
const RelatedVideosMock = RelatedVideos as jest.Mock;

describe("VideoDetail", () => {
  afterEach(() => {
    ChannelInfoMock.mockReset();
    RelatedVideosMock.mockReset();
  });

  it("renders video item details", () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<VideoDetail />} />, {
        pathname: "/",
        state: { video: fakeVideo },
        key: "fake-key",
      })
    );
    expect(component.toJSON()).toMatchSnapshot();

    const { channelId, channelTitle } = fakeVideo.snippet;
    expect(ChannelInfoMock.mock.calls[0][0]).toStrictEqual({
      id: channelId,
      name: channelTitle,
    });
    expect(RelatedVideosMock.mock.calls[0][0]).toStrictEqual({
      id: fakeVideo.id,
    });
  });
});
