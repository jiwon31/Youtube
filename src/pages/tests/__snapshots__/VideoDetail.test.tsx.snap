// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoDetail renders video item details 1`] = `
<section
  className="flex flex-col lg:flex-row"
>
  <article
    className="basis-4/6"
  >
    <iframe
      frameBorder="0"
      height="640"
      id="player"
      src="https://www.youtube.com/embed/1"
      title="title"
      width="100%"
    />
    <div
      className="p-8 pl-0"
    >
      <h2
        className="text-xl font-bold"
      >
        title
      </h2>
      <pre
        className="whitespace-pre-wrap"
      >
        description
      </pre>
    </div>
  </article>
  <section
    className="basis-2/6"
  />
</section>
`;
